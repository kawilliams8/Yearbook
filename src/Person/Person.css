.Person {
  background-color: white;
  border-radius: 6px;
  height: 500px;
  margin: 12px;
  width: 350px;
}

.name {
  color: blue;
  padding: 0px;
  display: block;
  font-size: 1.5em;
}

.photo {
  border-radius: 6px;
  height: 225px;
  object-fit: cover;
  width: 225px;
}

.quote {
  color: red;
  font-style: italic;
  padding: 0 30px;
}

.superlative {
  font-size: 12px;
}

.RemovePerson-button {
  background-color: white;
  color: red;
  font-size: 1.25em;
  height: 50px;
  margin: 12px;
  width: 200px;
}

.RemovePerson-button:hover {
  background-color: lightgrey;
  cursor: pointer;
}