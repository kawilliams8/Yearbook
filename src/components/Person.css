.Person {
  background-color: white;
  border-radius: 6px;
  height: 500px;
  margin: 12px;
  width: 350px;
}

.photo {
  border-radius: 6px;
  height: 250px;
  object-fit: cover;
  width: 250px;
}

.quote {
  color: red;
  font-style: italic;
  padding: 0 30px;
}

.superlative {
  font-size: 12px;
}