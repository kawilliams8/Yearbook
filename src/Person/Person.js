import React from 'react';
import './Person.css';

const Person = ({position, name, photo, quote, superlative, id, removeStudent}) => {
  return position === 'staff' ? 
    (
      <div className="Person">
        <h3 className="name">{name.toUpperCase()}</h3>
        <img src={photo} className="photo" alt={name}></img>
        <h4 className="quote">"{quote}"</h4>
        <h5 className="superlative">{superlative}</h5>
      </div>
    )
  :
    (
      <div className="Person">
        <h3 className="name" contentEditable={true} suppressContentEditableWarning="true">{name.toUpperCase()}</h3>
        <img src={photo} className="photo" alt={name}></img>
        <h4 className="quote" contentEditable={true} suppressContentEditableWarning="true">"{quote}"</h4>
        <h5 className="superlative" contentEditable={true} suppressContentEditableWarning="true">{superlative}</h5>
        <button className="deleteButton" onClick={() => removeStudent(id)}>DELETE</button>
      </div>
    )
}

export default Person;