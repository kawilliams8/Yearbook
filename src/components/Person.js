import React from 'react';
import './Person.css';

const Person = ({position, name, photo, quote, superlative, id, removeStudent}) => {
  if (position === 'staff') {
    return (
      <div className="Person">
        <h3 className="name">{name.toUpperCase()}</h3>
        <img src={photo} className="photo" alt={name}></img>
        <h4 className="quote">"{quote}"</h4>
        <h5 className="superlative">{superlative}</h5>
      </div>
    )
  } else {
    return (
      <div className="Person">
        <h3 className="name" contentEditable={true} >{name.toUpperCase()}</h3>
        <img src={photo} className="photo" alt={name}></img>
        <h4 className="quote" contentEditable={true} >"{quote}"</h4>
        <h5 className="superlative" contentEditable={true} >{superlative}</h5>
        <button className="deleteButton" onClick={() => removeStudent(id)}>DELETE</button>
      </div>
    )
  }
}

export default Person;