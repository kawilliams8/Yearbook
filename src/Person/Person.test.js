import React from 'react';
import { shallow } from 'enzyme';
import Person from './Person';

describe('Person', () => {
  it('', () => {

  });
});