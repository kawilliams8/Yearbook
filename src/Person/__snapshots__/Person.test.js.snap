// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should match the snapshot with all staff data passed in correctly 1`] = `
<div
  className="Person"
>
  <h3
    className="name"
  >
    ERIC
  </h3>
  <img
    alt="Eric"
    className="photo"
    src="./eric.jpg"
  />
  <h4
    className="quote"
  >
    "
    You miss all the shots you don't take.
    "
  </h4>
  <h5
    className="superlative"
  >
    Most Likely to Win the PGA
  </h5>
</div>
`;

exports[`Person should match the snapshot with all student data passed in correctly 1`] = `
<div
  className="Person"
>
  <h3
    className="name"
    contentEditable={true}
    suppressContentEditableWarning="true"
  >
    ROSA
  </h3>
  <img
    alt="Rosa"
    className="photo"
    src="https://placekitten.com/200/300"
  />
  <h4
    className="quote"
    contentEditable={true}
    suppressContentEditableWarning="true"
  >
    "
    Accept advice.
    "
  </h4>
  <h5
    className="superlative"
    contentEditable={true}
    suppressContentEditableWarning="true"
  >
    Most Likely to Write JavaScript
  </h5>
  <button
    className="deleteButton"
    onClick={[Function]}
  >
    DELETE
  </button>
</div>
`;
